helix-core/src/main/java/org/apache/helix/customizedstate/CustomizedStateProvider.java

```diff
@@ -113,7 +113,7 @@ public CustomizedState getCustomizedState(String customizedStateName, String res
   /**
    * Delete the customized state for a specified resource and a specified partition
    */
-  public void deletePerPartitionCustomizedState(String customizedStateName, String resourceName,
+  public synchronized void deletePerPartitionCustomizedState(String customizedStateName, String resourceName,
```

Comment:
Why we need the synchronized keyword at the first place? ZK should be the one who takes care of concurrent control, right?

Reply:
There can be multiple threads that try to delete partitions in a same resource. Do we want to leave the currency to the Zookeeper or maybe we can prevent it from the beginning. There are some similar pattern in the code base. We can discuss to decide whether we want to follow.

Reply:
ZK is the source of truth. Even we control the concurrency in one single node, we can not prevent multiple nodes concurrent access. So leverage ZK capability is the right way.
In detail, updater prevents unexpected overwrite. So I don't think the additional check is necessary. Unless we have a concrete counter-example.